### PR TITLE
Code sample and documentation for the distributed executor service

### PR DESCRIPTION
**Pull request by ihsandemir**
Added the code sample for the distributed executor service.

Added the README section for distributed execution service.

### CODE REVIEW

**Comment by ihsandemir** [comment]
verify-linux

File: README.md [hunk 1]
```diff
@@ -1968,9 +1978,283 @@ int main() {
...
+    }
+
+    virtual void onFailure(const boost::shared_ptr<exception::IException> &e) {
+        std::cout << "The execution of the task failed with exception:" << e << std::endl;
+    }
+};
+```
+
+#### 7.6.1.7 Selecting Members for Task Execution
+As previously mentioned, it is possible to indicate where in the Hazelcast cluster the task is executed. Usually you execute these in the cluster based on the location of a key or a set of keys, or you allow Hazelcast to select a member.
```
Comment (Serdaro) [inline README.md]: ... location of a key or set of keys ...

File: README.md [hunk 2]
```diff
@@ -1968,9 +1978,283 @@ int main() {
...
+
+If you want more control over where your code runs, use the `MemberSelector` interface. For example, you may want certain tasks to run only on certain members, or you may wish to implement some form of custom load balancing regime.  The `MemberSelector` is an interface that you can implement and then provide to the `IExecutorService` when you submit or execute.
+
+The `bool select(const Member &member)` method is called for every available member in the cluster. Implement this method to decide if the member is going to be used or not.
+
+In a simple example shown below, we select the cluster members based on the presence of an attribute.
```
Comment (Serdaro) [inline README.md]: In the simple example ...

File: README.md [hunk 3]
```diff
@@ -1968,9 +1978,283 @@ int main() {
...
+        return *attribute == "true";
+    }
+
+    virtual void toString(std::ostream &os) const {
+        os << "MyMemberSelector";
+    }
+};
+```
+
+You can now submit your task using this selector and the task will run on the member whose attribute key "my.special.executor" is set to string value "true". An example is shown below:
```
Comment (Serdaro) [inline README.md]: ... whose attribute key "my.special.executor" is set to "true". ...

File: README.md [hunk 4]
```diff
@@ -1968,9 +1978,283 @@ int main() {
...
+    HazelcastClient hz(clientConfig);
+
+    // Get the Distributed Executor Service
+    boost::shared_ptr<IExecutorService> ex = hz.getExecutorService("my-distributed-executor");
+    // Submit the MessagePrinter to a the provided Hazelcast Member
+    boost::shared_ptr<ICompletableFuture<std::string> > future = ex->submitToMember<MessagePrinter, std::string>(
+            MessagePrinter(input), member);
+
+    boost::shared_ptr<std::string> taskResult = future->get();
+````
```
Comment (Serdaro) [inline README.md]: Let's make this 3 ticks, otherwise it may break the formatting.

File: README.md [hunk 5]
```diff
@@ -1968,9 +1978,283 @@ int main() {
...
 ## 7.6. Distributed Computing
 
-This chapter explains how you can use Hazelcast IMDG's entry processor implementation in the C++ client.
+This section describes how Hazelcast IMDG's distributed executor service and entry processor features can be used in the C++ client.
 
-### 7.6.1. Using EntryProcessor
+### 7.6.1. Distributed Executor Service
+Hazelcast C++ client allows you to asynchronously execute your tasks (logical units of work) in the cluster, such as database queries, complex calculations and image rendering.
+
+With `IExecutorService`, you can execute tasks asynchronously and perform other useful tasks. If your task execution takes longer than expected, you can cancel the task execution. Tasks should be `Hazelcast Serializable`, i.e. `IdentifiedDataSerializable`, `Portable`, `Custom`, since they will be distributed in the cluster.
```
Comment (Serdaro) [inline README.md]: ... , i.e., ... (comma after i.e.)

**Pull request merged**